### ISSUE TITLE
Datepicker: focus and click don't work together

### ISSUE THREAD

**Issue by ctrl-brk** [valor-software/ngx-bootstrap#3698]
If you use [triggers]="'click focus'" then the first time you click on the datepicker it will open and close immediately. Seems like show() method works as a switch (not sure if this is desirable) and being called twice opens and closes the picker.

There are more issues with focus event. It will not assign selected date to the model when 'focus' used.

**Comment by IlyaSurmay** [comment]
This is a known issue, take a look at this - #3251

**Comment by ctrl-brk** [comment]
Yeah, but the other issue is closed. I was able to make it working using setTimeout for picker.hide(). There might be a better way but without both focus and click it's not user friendly.

**Comment by astalegale** [comment]
I quote, the problem on blur is relevant

**Comment by rkulinski** [comment]
I fully agree. It's kind of serious bug IMHO. If you 'tab out' of the input the datepicker doesn't hide. And having this kind of behaviour when loosing focus forces you to do not cool workarounds :(
